patches/tModLoader/Terraria.GameContent.UI.Chat/ItemTagHandler.cs.patch

```diff
@@ -0,0 +1,142 @@
+--- src/Terraria\Terraria.GameContent.UI.Chat\ItemTagHandler.cs
++++ src/tModLoader\Terraria.GameContent.UI.Chat\ItemTagHandler.cs
+@@ -2,6 +_,10 @@
+ using Microsoft.Xna.Framework.Graphics;
+ using ReLogic.Graphics;
+ using System;
++using System.Collections;
++using System.Collections.Generic;
++using System.Linq;
++using System.Text.RegularExpressions;
+ using Terraria.UI;
+ using Terraria.UI.Chat;
+ 
+@@ -9,7 +_,7 @@
+ {
+ 	public class ItemTagHandler : ITagHandler
+ 	{
+-		private class ItemSnippet : TextSnippet
++		internal class ItemSnippet : TextSnippet
+ 		{
+ 			private Item _item;
+ 
+@@ -22,10 +_,11 @@
+ 
+ 			public override void OnHover()
+ 			{
+-				Main.HoverItem = this._item.Clone();
++				Main.HoverItem = this._item;
+ 				Main.instance.MouseText(this._item.Name, this._item.rare, 0, -1, -1, -1, -1);
+ 			}
+ 
++			//@todo possibly allow modders to custom draw here
+ 			public override bool UniqueDraw(bool justCheckingString, out Vector2 size, SpriteBatch spriteBatch, Vector2 position = default(Vector2), Color color = default(Color), float scale = 1f)
+ 			{
+ 				float num = 1f;
+@@ -70,8 +_,50 @@
+ 			}
+ 		}
+ 
++		// stores item instances, used to properly display modded data
++		internal static readonly List<Item> Items = new List<Item>();
++
++		// attempts to remove stored item instances in Items
++		// that got pruned from the chat (i.e. they are no longer needed)
++		// in Main.cs at 60085 in NewText add this
++		// in Main.cs at 60043 in NewText add this
++		internal static bool TryRemoveLastStoredItem()
++		{
++			TextSnippet[] itemSnippets = Main.chatLine[Main.numChatLines - 1].parsedText
++				.Where(x => x.GetType() == typeof(ItemSnippet))
++				.ToArray();
++
++			if (itemSnippets.Any())
++			{
++				foreach (TextSnippet snippet in itemSnippets)
++				{
++					MatchCollection matchCollection = ChatManager.Regexes.Format.Matches(snippet.TextOriginal);
++					foreach (Match match in matchCollection)
++					{
++						string MIDGroup = match.Groups["options"].Value
```

Comment:
@Chicken-Bones I don't see the point. These is how vanilla does it, why make something new if this can already do what we want? (it's already a regex that does it)